common/src/main/java/com/microsoft/hydralab/common/file/impl/local/client/LocalStorageClient.java

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package com.microsoft.hydralab.common.file.impl.local.client;
+
+import com.alibaba.fastjson.JSONObject;
+import com.microsoft.hydralab.common.entity.common.StorageFileInfo;
+import com.microsoft.hydralab.common.file.impl.local.LocalStorageProperty;
+import com.microsoft.hydralab.common.file.impl.local.LocalStorageToken;
+import com.microsoft.hydralab.common.util.Const;
+import com.microsoft.hydralab.common.util.HydraLabRuntimeException;
+import com.microsoft.hydralab.common.util.RestTemplateConfig;
+import lombok.Data;
+import org.apache.commons.io.IOUtils;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+/**
+ * @author Li Shen
+ * @date 3/6/2023
+ */
+
+@Data
+public class LocalStorageClient {
+    RestTemplate restTemplate;
+    private String endpoint;
+    private String token;
+
+    public LocalStorageClient() {
+    }
+
+    public LocalStorageClient(LocalStorageProperty localStorageProperty) {
+        this.endpoint = localStorageProperty.getEndpoint();
+        if (this.endpoint.endsWith("/")) {
+            this.endpoint = this.endpoint.substring(0, this.endpoint.length() - 1);
+        }
+        this.token = localStorageProperty.getToken();
+    }
+
+    public void init(LocalStorageToken localStorageToken) {
```

Comment:
Why we need this init? It seems it could be replaced with the constructor.

Reply:
Updated.